railib/api.py

```diff
@@ -325,6 +326,38 @@ def _parse_arrow_results(files: List[TransactionAsyncFile]):
             results.append({"relationId": file.name, "table": table})
     return results
 
+# polling with specified overhead
+# delay is the overhead % of the time the transaction has been running so far
+
+
+def poll_with_specified_overhead(
+    f,
+    overhead_rate: float = 0.1,
+    start_time: datetime = datetime.now(timezone.utc),
+    timeout: int = None,
+    max_retries: int = None
```

Comment:
`max_retries` is never used by any tests or examples. Speaking of which, why do we need `max_retries` when we have `timeout`?

Reply:
this was inspired from the julia sdk https://github.com/RelationalAI/rai-sdk-julia/blob/main/src/api.jl#L126-L157
`timeout` and 'max_retries` could/should co-exist a polling library for python do the same
https://github.com/justiniso/polling/blob/master/polling.py#L44

Reply:
Then add tests. Also, 'retry' is usually something you do when you get an error. `max_tries` from the polling library seems more correct.